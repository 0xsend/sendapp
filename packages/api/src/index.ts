export * from './routers/_app'
export * from './trpc'
