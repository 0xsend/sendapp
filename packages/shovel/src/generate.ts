import { toJSON } from '@indexsupply/shovel-config'
import { config } from '.'
import { parseArgs } from 'node:util'

const { values } = parseArgs({
  args: Bun.argv,
  options: {
    path: {
      type: 'string',
      default: 'shovel-config.json',
    },
  },
  strict: true,
  allowPositionals: true,
})

if (!values.path) {
  throw new Error('path is required')
}

Bun.write(values.path, toJSON(config, 2))
