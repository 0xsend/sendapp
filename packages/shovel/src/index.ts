import { makeConfig } from '@indexsupply/shovel-config'
import type { Source, Integration } from '@indexsupply/shovel-config'
import {
  sendAccountCreatedIntegration,
  sendAccountTransfersIntegration,
  sendAccountReceivesIntegration,
  sendTokenTransfersIntegration,
  sendRevenuesSafeReceives,
  sendAccountSigningKeyAdded,
  sendAccountSigningKeyRemoved,
} from './integrations'

// baseSrcBlockHeaders is to be used for integrations that require block headers
const baseSrcBlockHeaders: Source = {
  name: 'base_block_headers',
  url: '$BASE_RPC_URL',
  urls: ['$BASE_RPC_URL'],
  chain_id: '$BASE_CHAIN_ID',
  batch_size: 100,
  concurrency: 1,
}

// baseSrcLogs is to be used for integrations that require logs
const baseSrcLogs: Source = {
  name: 'base_logs',
  url: '$BASE_RPC_URL',
  urls: ['$BASE_RPC_URL'],
  chain_id: '$BASE_CHAIN_ID',
  batch_size: 2000,
  concurrency: 2,
}

export const sources: Source[] = [baseSrcBlockHeaders, baseSrcLogs]

export const integrations: Integration[] = [
  {
    ...sendAccountCreatedIntegration,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
  {
    ...sendAccountTransfersIntegration,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
  {
    ...sendTokenTransfersIntegration,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
  {
    ...sendRevenuesSafeReceives,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
  {
    ...sendAccountSigningKeyAdded,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
  {
    ...sendAccountSigningKeyRemoved,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
  {
    ...sendAccountReceivesIntegration,
    sources: [{ name: baseSrcLogs.name, start: '$BASE_BLOCK_START' }],
  },
]

const c = makeConfig({
  pg_url: '$DATABASE_URL',
  sources,
  integrations,
  dashboard: {
    root_password: '$DASHBOARD_ROOT_PASSWORD',
  },
})

export const config = c
