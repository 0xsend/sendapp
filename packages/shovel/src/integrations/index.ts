export { integration as sendAccountCreatedIntegration } from './send-account-created'
export { integration as sendAccountTransfersIntegration } from './send-account-transfers'
export { integration as sendAccountTransactionsIntegration } from './send-account-transactions'
export { integration as sendTokenTransfersIntegration } from './send-token-transfers'
export { integration as sendRevenuesSafeReceives } from './send-revenues-safe-receives'
export { integration as sendAccountSigningKeyAdded } from './send-account-signing-key-added'
export { integration as sendAccountSigningKeyRemoved } from './send-account-signing-key-removed'
