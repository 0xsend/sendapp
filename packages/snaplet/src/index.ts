import type { Store } from '@snaplet/seed'

export type { SeedClient, Store } from '@snaplet/seed'
export type StoreScalar<K extends keyof Store> = Store[K][number]

export { tagName } from './utils'
export * from './models'
export {
  createUserWithConfirmedTags,
  createUserWithTagsAndAccounts,
  createMultipleUsersWithTags,
} from './models'
