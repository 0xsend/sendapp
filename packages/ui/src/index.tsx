export * from '@tamagui/toast'
export * from 'tamagui'
export * from './components'
export { config } from './tamagui.config'
