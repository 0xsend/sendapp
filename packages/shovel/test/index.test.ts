import { config } from '../src'
import { test, expect } from 'bun:test'

test('shovel config', () => {
  expect(config).toMatchSnapshot()
})
