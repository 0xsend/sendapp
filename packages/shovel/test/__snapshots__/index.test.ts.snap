// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`shovel config 1`] = `
{
  "dashboard": {
    "root_password": "$DASHBOARD_ROOT_PASSWORD",
  },
  "integrations": [
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x008c9561857b6555584d20aC55110335759Aa2c2",
            "0xe4C5EF95e8cDA5DB09393a08E30645F883e187B8",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "account",
            "indexed": true,
            "name": "account",
            "type": "address",
          },
        ],
        "name": "AccountCreated",
        "type": "event",
      },
      "name": "send_account_created",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "user_op_hash",
            "type": "bytea",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "account",
            "type": "bytea",
          },
        ],
        "name": "send_account_created",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x036CbD53842c5426634e7929541eC2318f3dCF7e",
            "0x08210F9170F89Ab7658F0B5E3fF39b0E03C594D4",
            "0x08E53B71490E00e8dC1c0367f97BA053567a547E",
            "0x1aBaEA1f7C830bD89Acc67eC4af516284b1bC33c",
            "0x1c7D4B196Cb0C7B01d743Fbc6116a902379C7238",
            "0x3f14920c99BEB920Afa163031c4e47a3e03B3e4A",
            "0x50dA645f148798F68EF2d7dB7C1CB22A6819bb2C",
            "0x58D97B57BB95320F9a05dC918Aef65434969c2B2",
            "0x60a3E35Cc302bFA44Cb288Bc5a4F316Fdb1adb42",
            "0x7cEfbe54c37a35dCdaD29b86373ca8353a2F4680",
            "0x808456652fdb597867f38412077A9182bf77359F",
            "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
            "0x940181a94A35A4569E4529A3CDfB74e38FD98631",
            "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
            "0xA88594D404727625A9437C3f886C7643872296AE",
            "0xBAa5CC21fd487B8Fcc2F632f3F4E8D37262a0842",
            "0xBbB542c66a7DD7BA6893C9630B30358D610FF3ee",
            "0xE0f63A424a4439cBE457D80E4f4b51aD25b2c56C",
            "0xEab49138BA2Ea6dd776220fE26b7b8E446638956",
            "0xcbB7C0000aB88B473b1f5aFd9ef808440eed33Bf",
            "0xcbB7C0006F23900c38EB856149F799620fcb8A4a",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "f",
            "indexed": true,
            "name": "from",
            "type": "address",
          },
          {
            "column": "t",
            "indexed": true,
            "name": "to",
            "type": "address",
          },
          {
            "column": "v",
            "indexed": false,
            "name": "value",
            "type": "uint256",
          },
        ],
        "name": "Transfer",
        "type": "event",
      },
      "name": "send_account_transfers",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "f",
            "type": "bytea",
          },
          {
            "name": "t",
            "type": "bytea",
          },
          {
            "name": "v",
            "type": "numeric",
          },
        ],
        "name": "send_account_transfers",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0xBbB542c66a7DD7BA6893C9630B30358D610FF3ee",
            "0xEab49138BA2Ea6dd776220fE26b7b8E446638956",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "f",
            "indexed": true,
            "name": "from",
            "type": "address",
          },
          {
            "column": "t",
            "indexed": true,
            "name": "to",
            "type": "address",
          },
          {
            "column": "v",
            "indexed": false,
            "name": "value",
            "type": "uint256",
          },
        ],
        "name": "Transfer",
        "type": "event",
      },
      "name": "send_token_transfers",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "f",
            "type": "bytea",
          },
          {
            "name": "t",
            "type": "bytea",
          },
          {
            "name": "v",
            "type": "numeric",
          },
        ],
        "name": "send_token_transfers",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x3f14920c99BEB920Afa163031c4e47a3e03B3e4A",
            "0x7cEfbe54c37a35dCdaD29b86373ca8353a2F4680",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "f",
            "indexed": true,
            "name": "from",
            "type": "address",
          },
          {
            "column": "t",
            "indexed": true,
            "name": "to",
            "type": "address",
          },
          {
            "column": "v",
            "indexed": false,
            "name": "value",
            "type": "uint256",
          },
        ],
        "name": "Transfer",
        "type": "event",
      },
      "name": "send_token_v0_transfers",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "f",
            "type": "bytea",
          },
          {
            "name": "t",
            "type": "bytea",
          },
          {
            "name": "v",
            "type": "numeric",
          },
        ],
        "name": "send_token_v0_transfers",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x269cD0a2afd1BAbdA7A74ab1dC853869a37aa4a7",
            "0x71fa02bb11e4b119bEDbeeD2f119F62048245301",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "inputs": [
          {
            "column": "sender",
            "indexed": true,
            "internalType": "address",
            "name": "sender",
            "type": "address",
          },
          {
            "column": "v",
            "indexed": false,
            "name": "value",
            "type": "uint256",
          },
        ],
        "name": "SafeReceived",
        "type": "event",
      },
      "name": "send_revenues_safe_receives",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "sender",
            "type": "bytea",
          },
          {
            "name": "v",
            "type": "numeric",
          },
        ],
        "name": "send_revenues_safe_receives",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_op": "contains",
          "filter_ref": {
            "column": "account",
            "integration": "send_account_created",
          },
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "account",
            "indexed": true,
            "name": "account",
            "type": "address",
          },
          {
            "column": "key_slot",
            "indexed": false,
            "name": "keySlot",
            "type": "uint8",
          },
          {
            "column": "key",
            "indexed": false,
            "name": "key",
            "type": "bytes32[2]",
          },
        ],
        "name": "SigningKeyAdded",
        "type": "event",
      },
      "name": "send_account_signing_key_added",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "account",
            "type": "bytea",
          },
          {
            "name": "key_slot",
            "type": "smallint",
          },
          {
            "name": "key",
            "type": "bytea",
          },
        ],
        "name": "send_account_signing_key_added",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_op": "contains",
          "filter_ref": {
            "column": "account",
            "integration": "send_account_created",
          },
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "account",
            "indexed": true,
            "name": "account",
            "type": "address",
          },
          {
            "column": "key_slot",
            "indexed": false,
            "name": "keySlot",
            "type": "uint8",
          },
          {
            "column": "key",
            "indexed": false,
            "name": "key",
            "type": "bytes32[2]",
          },
        ],
        "name": "SigningKeyRemoved",
        "type": "event",
      },
      "name": "send_account_signing_key_removed",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "account",
            "type": "bytea",
          },
          {
            "name": "key_slot",
            "type": "smallint",
          },
          {
            "name": "key",
            "type": "bytea",
          },
        ],
        "name": "send_account_signing_key_removed",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_num",
          "name": "block_num",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "tx_idx",
          "name": "tx_idx",
        },
        {
          "column": "log_idx",
          "name": "log_idx",
        },
        {
          "column": "log_addr",
          "filter_op": "contains",
          "filter_ref": {
            "column": "account",
            "integration": "send_account_created",
          },
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "sender",
            "filter_op": "contains",
            "filter_ref": {
              "column": "account",
              "integration": "send_account_created",
            },
            "indexed": true,
            "internalType": "address",
            "name": "sender",
            "type": "address",
          },
          {
            "column": "value",
            "indexed": false,
            "internalType": "uint256",
            "name": "value",
            "type": "uint256",
          },
        ],
        "name": "Received",
        "type": "event",
      },
      "name": "send_account_receives",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "block_num",
            "type": "numeric",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "tx_idx",
            "type": "numeric",
          },
          {
            "name": "log_idx",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "sender",
            "type": "bytea",
          },
          {
            "name": "value",
            "type": "numeric",
          },
        ],
        "name": "send_account_receives",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x36f43082d01df4801AF2D95aeEd1a0200C5510AE",
            "0x32b6F71420a0f8286D10D83eAf9EB8F12e6f33Ca",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "sender",
            "indexed": true,
            "internalType": "address",
            "name": "sender",
            "type": "address",
          },
          {
            "column": "amount",
            "indexed": false,
            "internalType": "uint256",
            "name": "amount",
            "type": "uint256",
          },
          {
            "column": "referrer",
            "indexed": false,
            "internalType": "address",
            "name": "referrer",
            "type": "address",
          },
          {
            "column": "reward",
            "indexed": false,
            "internalType": "uint256",
            "name": "reward",
            "type": "uint256",
          },
        ],
        "name": "Receipt",
        "type": "event",
      },
      "name": "sendtag_checkout_receipts",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "sender",
            "type": "bytea",
          },
          {
            "name": "amount",
            "type": "numeric",
          },
          {
            "name": "referrer",
            "type": "bytea",
          },
          {
            "name": "reward",
            "type": "numeric",
          },
        ],
        "name": "sendtag_checkout_receipts",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_num",
          "name": "block_num",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "tx_idx",
          "name": "tx_idx",
        },
        {
          "column": "log_idx",
          "name": "log_idx",
        },
        {
          "column": "log_addr",
          "filter_op": "equals",
          "filter_val": "0xa0A5611b9A1071a1D8A308882065c48650bAeE8b",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "recipient",
            "indexed": true,
            "internalType": "address",
            "name": "recipient",
            "type": "address",
          },
          {
            "column": "tickets_purchased_total_bps",
            "indexed": false,
            "internalType": "uint256",
            "name": "ticketsPurchasedTotalBps",
            "type": "uint256",
          },
          {
            "column": "referrer",
            "indexed": true,
            "internalType": "address",
            "name": "referrer",
            "type": "address",
          },
          {
            "column": "buyer",
            "indexed": true,
            "internalType": "address",
            "name": "buyer",
            "type": "address",
          },
        ],
        "name": "UserTicketPurchase",
        "type": "event",
      },
      "name": "sendpot_user_ticket_purchases",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "constraints": "PRIMARY KEY",
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "constraints": "PRIMARY KEY",
            "name": "block_num",
            "type": "numeric",
          },
          {
            "constraints": "PRIMARY KEY",
            "name": "log_idx",
            "type": "numeric",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "tx_idx",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "recipient",
            "type": "bytea",
          },
          {
            "name": "tickets_purchased_total_bps",
            "type": "numeric",
          },
          {
            "name": "referrer",
            "type": "bytea",
          },
          {
            "name": "buyer",
            "type": "bytea",
          },
        ],
        "name": "sendpot_user_ticket_purchases",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_num",
          "name": "block_num",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "tx_idx",
          "name": "tx_idx",
        },
        {
          "column": "log_idx",
          "name": "log_idx",
        },
        {
          "column": "log_addr",
          "filter_op": "equals",
          "filter_val": "0xa0A5611b9A1071a1D8A308882065c48650bAeE8b",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "time",
            "indexed": false,
            "internalType": "uint256",
            "name": "time",
            "type": "uint256",
          },
          {
            "column": "winner",
            "indexed": false,
            "internalType": "address",
            "name": "winner",
            "type": "address",
          },
          {
            "column": "winning_ticket",
            "indexed": false,
            "internalType": "uint256",
            "name": "winningTicket",
            "type": "uint256",
          },
          {
            "column": "win_amount",
            "indexed": false,
            "internalType": "uint256",
            "name": "winAmount",
            "type": "uint256",
          },
          {
            "column": "tickets_purchased_total_bps",
            "indexed": false,
            "internalType": "uint256",
            "name": "ticketsPurchasedTotalBps",
            "type": "uint256",
          },
        ],
        "name": "JackpotRun",
        "type": "event",
      },
      "name": "sendpot_jackpot_runs",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "constraints": "PRIMARY KEY",
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "constraints": "PRIMARY KEY",
            "name": "block_num",
            "type": "numeric",
          },
          {
            "constraints": "PRIMARY KEY",
            "name": "log_idx",
            "type": "numeric",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "tx_idx",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "time",
            "type": "numeric",
          },
          {
            "name": "winner",
            "type": "bytea",
          },
          {
            "name": "winning_ticket",
            "type": "numeric",
          },
          {
            "name": "win_amount",
            "type": "numeric",
          },
          {
            "name": "tickets_purchased_total_bps",
            "type": "numeric",
          },
        ],
        "name": "sendpot_jackpot_runs",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x392F337911A610a3850bfE7805ab9C948B836838",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "send_earn",
            "indexed": true,
            "name": "sendEarn",
            "type": "address",
          },
          {
            "column": "caller",
            "indexed": true,
            "name": "caller",
            "type": "address",
          },
          {
            "column": "initial_owner",
            "indexed": false,
            "name": "initialOwner",
            "type": "address",
          },
          {
            "column": "vault",
            "indexed": true,
            "name": "vault",
            "type": "address",
          },
          {
            "column": "fee_recipient",
            "indexed": false,
            "name": "feeRecipient",
            "type": "address",
          },
          {
            "column": "collections",
            "indexed": false,
            "name": "collections",
            "type": "address",
          },
          {
            "column": "fee",
            "indexed": false,
            "name": "fee",
            "type": "uint96",
          },
          {
            "column": "salt",
            "indexed": false,
            "name": "salt",
            "type": "bytes32",
          },
        ],
        "name": "CreateSendEarn",
        "type": "event",
      },
      "name": "send_earn_create",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "send_earn",
            "type": "bytea",
          },
          {
            "name": "caller",
            "type": "bytea",
          },
          {
            "name": "initial_owner",
            "type": "bytea",
          },
          {
            "name": "vault",
            "type": "bytea",
          },
          {
            "name": "fee_recipient",
            "type": "bytea",
          },
          {
            "name": "collections",
            "type": "bytea",
          },
          {
            "name": "fee",
            "type": "numeric",
          },
          {
            "name": "salt",
            "type": "bytea",
          },
        ],
        "name": "send_earn_create",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "filter_arg": [
            "0x392F337911A610a3850bfE7805ab9C948B836838",
          ],
          "filter_op": "contains",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "affiliate",
            "indexed": false,
            "name": "affiliate",
            "type": "address",
          },
          {
            "column": "send_earn_affiliate",
            "indexed": false,
            "name": "send_earn_affiliate",
            "type": "address",
          },
        ],
        "name": "NewAffiliate",
        "type": "event",
      },
      "name": "send_earn_new_affiliate",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "affiliate",
            "type": "bytea",
          },
          {
            "name": "send_earn_affiliate",
            "type": "bytea",
          },
        ],
        "name": "send_earn_new_affiliate",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "sender",
            "indexed": true,
            "name": "sender",
            "type": "address",
          },
          {
            "column": "owner",
            "indexed": true,
            "name": "owner",
            "type": "address",
          },
          {
            "column": "assets",
            "indexed": false,
            "name": "assets",
            "type": "uint256",
          },
          {
            "column": "shares",
            "indexed": false,
            "name": "shares",
            "type": "uint256",
          },
        ],
        "name": "Deposit",
        "type": "event",
      },
      "name": "send_earn_deposit",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "sender",
            "type": "bytea",
          },
          {
            "name": "owner",
            "type": "bytea",
          },
          {
            "name": "assets",
            "type": "numeric",
          },
          {
            "name": "shares",
            "type": "numeric",
          },
        ],
        "name": "send_earn_deposit",
      },
    },
    {
      "block": [
        {
          "column": "chain_id",
          "name": "chain_id",
        },
        {
          "column": "block_time",
          "name": "block_time",
        },
        {
          "column": "tx_hash",
          "name": "tx_hash",
        },
        {
          "column": "log_addr",
          "name": "log_addr",
        },
      ],
      "enabled": true,
      "event": {
        "anonymous": false,
        "inputs": [
          {
            "column": "sender",
            "indexed": true,
            "name": "sender",
            "type": "address",
          },
          {
            "column": "receiver",
            "indexed": true,
            "name": "receiver",
            "type": "address",
          },
          {
            "column": "owner",
            "indexed": true,
            "name": "owner",
            "type": "address",
          },
          {
            "column": "assets",
            "indexed": false,
            "name": "assets",
            "type": "uint256",
          },
          {
            "column": "shares",
            "indexed": false,
            "name": "shares",
            "type": "uint256",
          },
        ],
        "name": "Withdraw",
        "type": "event",
      },
      "name": "send_earn_withdraw",
      "sources": [
        {
          "name": "base_logs",
          "start": "$BASE_BLOCK_START",
        },
      ],
      "table": {
        "columns": [
          {
            "name": "chain_id",
            "type": "numeric",
          },
          {
            "name": "log_addr",
            "type": "bytea",
          },
          {
            "name": "block_time",
            "type": "numeric",
          },
          {
            "name": "tx_hash",
            "type": "bytea",
          },
          {
            "name": "sender",
            "type": "bytea",
          },
          {
            "name": "receiver",
            "type": "bytea",
          },
          {
            "name": "owner",
            "type": "bytea",
          },
          {
            "name": "assets",
            "type": "numeric",
          },
          {
            "name": "shares",
            "type": "numeric",
          },
        ],
        "name": "send_earn_withdraw",
      },
    },
  ],
  "pg_url": "$DATABASE_URL",
  "sources": [
    {
      "batch_size": 100,
      "chain_id": "$BASE_CHAIN_ID",
      "concurrency": 1,
      "name": "base_block_headers",
      "url": "$BASE_RPC_URL_PRIMARY",
      "urls": [
        "$BASE_RPC_URL_PRIMARY",
        "$BASE_RPC_URL_BACKUP1",
        "$BASE_RPC_URL_PRIMARY",
        "$BASE_RPC_URL_BACKUP2",
        "$BASE_RPC_URL_PRIMARY",
        "$BASE_RPC_URL_BACKUP3",
      ],
    },
    {
      "batch_size": 2000,
      "chain_id": "$BASE_CHAIN_ID",
      "concurrency": 2,
      "name": "base_logs",
      "url": "$BASE_RPC_URL_PRIMARY",
      "urls": [
        "$BASE_RPC_URL_PRIMARY",
        "$BASE_RPC_URL_BACKUP1",
        "$BASE_RPC_URL_PRIMARY",
        "$BASE_RPC_URL_BACKUP2",
        "$BASE_RPC_URL_PRIMARY",
        "$BASE_RPC_URL_BACKUP3",
      ],
    },
  ],
}
`;
